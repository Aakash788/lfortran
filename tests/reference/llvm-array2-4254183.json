{
    "basename": "llvm-array2-4254183",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/array2.f90",
    "infile_hash": "ec67f153c5f506f6a12ef94b06d43c7e9e86b9fc33c9c196029183f0",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-array2-4254183.stdout",
    "stdout_hash": "74d558a340444eed7bd922c00f249c27df6b8ec51a7d31642f51cb28",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}