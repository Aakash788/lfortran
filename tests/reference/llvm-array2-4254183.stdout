; ModuleID = 'LFortran'
source_filename = "LFortran"

%array = type { float*, i32, %dimension_descriptor*, i1 }
%dimension_descriptor = type { i32, i32, i32, i32 }
%array.0 = type { i32*, i32, %dimension_descriptor*, i1 }
%array.1 = type { i1*, i32, %dimension_descriptor*, i1 }
%array.2 = type { float*, i32, %dimension_descriptor*, i1 }
%array.3 = type { i32*, i32, %dimension_descriptor*, i1 }
%array.4 = type { i1*, i32, %dimension_descriptor*, i1 }
%array.5 = type { float*, i32, %dimension_descriptor*, i1 }
%array.6 = type { i32*, i32, %dimension_descriptor*, i1 }
%array.7 = type { i1*, i32, %dimension_descriptor*, i1 }

define i32 @main() {
.entry:
  %a = alloca %array, align 8
  %0 = getelementptr %array, %array* %a, i32 0, i32 1
  store i32 0, i32* %0, align 4
  %1 = getelementptr %array, %array* %a, i32 0, i32 2
  %2 = alloca i32, align 4
  store i32 1, i32* %2, align 4
  %3 = load i32, i32* %2, align 4
  %4 = alloca %dimension_descriptor, i32 %3, align 8
  store %dimension_descriptor* %4, %dimension_descriptor** %1, align 8
  %5 = load %dimension_descriptor*, %dimension_descriptor** %1, align 8
  %6 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %5, i32 0
  %7 = getelementptr %dimension_descriptor, %dimension_descriptor* %6, i32 0, i32 0
  %8 = getelementptr %dimension_descriptor, %dimension_descriptor* %6, i32 0, i32 1
  %9 = getelementptr %dimension_descriptor, %dimension_descriptor* %6, i32 0, i32 2
  %10 = getelementptr %dimension_descriptor, %dimension_descriptor* %6, i32 0, i32 3
  store i32 1, i32* %7, align 4
  store i32 1, i32* %8, align 4
  store i32 5, i32* %9, align 4
  %11 = load i32, i32* %9, align 4
  %12 = load i32, i32* %8, align 4
  %13 = sub i32 %11, %12
  %14 = add i32 %13, 1
  store i32 %14, i32* %10, align 4
  %15 = alloca i32, align 4
  store i32 5, i32* %15, align 4
  %16 = getelementptr %array, %array* %a, i32 0, i32 0
  %17 = load i32, i32* %15, align 4
  %18 = alloca float, i32 %17, align 4
  store float* %18, float** %16, align 8
  %b = alloca %array, align 8
  %19 = getelementptr %array, %array* %b, i32 0, i32 1
  store i32 0, i32* %19, align 4
  %20 = getelementptr %array, %array* %b, i32 0, i32 2
  %21 = alloca i32, align 4
  store i32 1, i32* %21, align 4
  %22 = load i32, i32* %21, align 4
  %23 = alloca %dimension_descriptor, i32 %22, align 8
  store %dimension_descriptor* %23, %dimension_descriptor** %20, align 8
  %24 = load %dimension_descriptor*, %dimension_descriptor** %20, align 8
  %25 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %24, i32 0
  %26 = getelementptr %dimension_descriptor, %dimension_descriptor* %25, i32 0, i32 0
  %27 = getelementptr %dimension_descriptor, %dimension_descriptor* %25, i32 0, i32 1
  %28 = getelementptr %dimension_descriptor, %dimension_descriptor* %25, i32 0, i32 2
  %29 = getelementptr %dimension_descriptor, %dimension_descriptor* %25, i32 0, i32 3
  store i32 1, i32* %26, align 4
  store i32 1, i32* %27, align 4
  store i32 5, i32* %28, align 4
  %30 = load i32, i32* %28, align 4
  %31 = load i32, i32* %27, align 4
  %32 = sub i32 %30, %31
  %33 = add i32 %32, 1
  store i32 %33, i32* %29, align 4
  %34 = alloca i32, align 4
  store i32 5, i32* %34, align 4
  %35 = getelementptr %array, %array* %b, i32 0, i32 0
  %36 = load i32, i32* %34, align 4
  %37 = alloca float, i32 %36, align 4
  store float* %37, float** %35, align 8
  %c = alloca %array.0, align 8
  %38 = getelementptr %array.0, %array.0* %c, i32 0, i32 1
  store i32 0, i32* %38, align 4
  %39 = getelementptr %array.0, %array.0* %c, i32 0, i32 2
  %40 = alloca i32, align 4
  store i32 1, i32* %40, align 4
  %41 = load i32, i32* %40, align 4
  %42 = alloca %dimension_descriptor, i32 %41, align 8
  store %dimension_descriptor* %42, %dimension_descriptor** %39, align 8
  %43 = load %dimension_descriptor*, %dimension_descriptor** %39, align 8
  %44 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %43, i32 0
  %45 = getelementptr %dimension_descriptor, %dimension_descriptor* %44, i32 0, i32 0
  %46 = getelementptr %dimension_descriptor, %dimension_descriptor* %44, i32 0, i32 1
  %47 = getelementptr %dimension_descriptor, %dimension_descriptor* %44, i32 0, i32 2
  %48 = getelementptr %dimension_descriptor, %dimension_descriptor* %44, i32 0, i32 3
  store i32 1, i32* %45, align 4
  store i32 1, i32* %46, align 4
  store i32 3, i32* %47, align 4
  %49 = load i32, i32* %47, align 4
  %50 = load i32, i32* %46, align 4
  %51 = sub i32 %49, %50
  %52 = add i32 %51, 1
  store i32 %52, i32* %48, align 4
  %53 = alloca i32, align 4
  store i32 3, i32* %53, align 4
  %54 = getelementptr %array.0, %array.0* %c, i32 0, i32 0
  %55 = load i32, i32* %53, align 4
  %56 = alloca i32, i32 %55, align 4
  store i32* %56, i32** %54, align 8
  %d = alloca %array.1, align 8
  %57 = getelementptr %array.1, %array.1* %d, i32 0, i32 1
  store i32 0, i32* %57, align 4
  %58 = getelementptr %array.1, %array.1* %d, i32 0, i32 2
  %59 = alloca i32, align 4
  store i32 1, i32* %59, align 4
  %60 = load i32, i32* %59, align 4
  %61 = alloca %dimension_descriptor, i32 %60, align 8
  store %dimension_descriptor* %61, %dimension_descriptor** %58, align 8
  %62 = load %dimension_descriptor*, %dimension_descriptor** %58, align 8
  %63 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %62, i32 0
  %64 = getelementptr %dimension_descriptor, %dimension_descriptor* %63, i32 0, i32 0
  %65 = getelementptr %dimension_descriptor, %dimension_descriptor* %63, i32 0, i32 1
  %66 = getelementptr %dimension_descriptor, %dimension_descriptor* %63, i32 0, i32 2
  %67 = getelementptr %dimension_descriptor, %dimension_descriptor* %63, i32 0, i32 3
  store i32 1, i32* %64, align 4
  store i32 1, i32* %65, align 4
  store i32 2, i32* %66, align 4
  %68 = load i32, i32* %66, align 4
  %69 = load i32, i32* %65, align 4
  %70 = sub i32 %68, %69
  %71 = add i32 %70, 1
  store i32 %71, i32* %67, align 4
  %72 = alloca i32, align 4
  store i32 2, i32* %72, align 4
  %73 = getelementptr %array.1, %array.1* %d, i32 0, i32 0
  %74 = load i32, i32* %72, align 4
  %75 = alloca i1, i32 %74, align 1
  store i1* %75, i1** %73, align 8
  %e = alloca %array.2, align 8
  %76 = getelementptr %array.2, %array.2* %e, i32 0, i32 1
  store i32 0, i32* %76, align 4
  %77 = getelementptr %array.2, %array.2* %e, i32 0, i32 2
  %78 = alloca i32, align 4
  store i32 2, i32* %78, align 4
  %79 = load i32, i32* %78, align 4
  %80 = alloca %dimension_descriptor, i32 %79, align 8
  store %dimension_descriptor* %80, %dimension_descriptor** %77, align 8
  %81 = load %dimension_descriptor*, %dimension_descriptor** %77, align 8
  %82 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %81, i32 0
  %83 = getelementptr %dimension_descriptor, %dimension_descriptor* %82, i32 0, i32 0
  %84 = getelementptr %dimension_descriptor, %dimension_descriptor* %82, i32 0, i32 1
  %85 = getelementptr %dimension_descriptor, %dimension_descriptor* %82, i32 0, i32 2
  %86 = getelementptr %dimension_descriptor, %dimension_descriptor* %82, i32 0, i32 3
  store i32 1, i32* %83, align 4
  store i32 1, i32* %84, align 4
  store i32 2, i32* %85, align 4
  %87 = load i32, i32* %85, align 4
  %88 = load i32, i32* %84, align 4
  %89 = sub i32 %87, %88
  %90 = add i32 %89, 1
  store i32 %90, i32* %86, align 4
  %91 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %81, i32 1
  %92 = getelementptr %dimension_descriptor, %dimension_descriptor* %91, i32 0, i32 0
  %93 = getelementptr %dimension_descriptor, %dimension_descriptor* %91, i32 0, i32 1
  %94 = getelementptr %dimension_descriptor, %dimension_descriptor* %91, i32 0, i32 2
  %95 = getelementptr %dimension_descriptor, %dimension_descriptor* %91, i32 0, i32 3
  store i32 1, i32* %92, align 4
  store i32 1, i32* %93, align 4
  store i32 3, i32* %94, align 4
  %96 = load i32, i32* %94, align 4
  %97 = load i32, i32* %93, align 4
  %98 = sub i32 %96, %97
  %99 = add i32 %98, 1
  store i32 %99, i32* %95, align 4
  %100 = alloca i32, align 4
  store i32 6, i32* %100, align 4
  %101 = getelementptr %array.2, %array.2* %e, i32 0, i32 0
  %102 = load i32, i32* %100, align 4
  %103 = alloca float, i32 %102, align 4
  store float* %103, float** %101, align 8
  %f = alloca %array.3, align 8
  %104 = getelementptr %array.3, %array.3* %f, i32 0, i32 1
  store i32 0, i32* %104, align 4
  %105 = getelementptr %array.3, %array.3* %f, i32 0, i32 2
  %106 = alloca i32, align 4
  store i32 2, i32* %106, align 4
  %107 = load i32, i32* %106, align 4
  %108 = alloca %dimension_descriptor, i32 %107, align 8
  store %dimension_descriptor* %108, %dimension_descriptor** %105, align 8
  %109 = load %dimension_descriptor*, %dimension_descriptor** %105, align 8
  %110 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %109, i32 0
  %111 = getelementptr %dimension_descriptor, %dimension_descriptor* %110, i32 0, i32 0
  %112 = getelementptr %dimension_descriptor, %dimension_descriptor* %110, i32 0, i32 1
  %113 = getelementptr %dimension_descriptor, %dimension_descriptor* %110, i32 0, i32 2
  %114 = getelementptr %dimension_descriptor, %dimension_descriptor* %110, i32 0, i32 3
  store i32 1, i32* %111, align 4
  store i32 1, i32* %112, align 4
  store i32 3, i32* %113, align 4
  %115 = load i32, i32* %113, align 4
  %116 = load i32, i32* %112, align 4
  %117 = sub i32 %115, %116
  %118 = add i32 %117, 1
  store i32 %118, i32* %114, align 4
  %119 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %109, i32 1
  %120 = getelementptr %dimension_descriptor, %dimension_descriptor* %119, i32 0, i32 0
  %121 = getelementptr %dimension_descriptor, %dimension_descriptor* %119, i32 0, i32 1
  %122 = getelementptr %dimension_descriptor, %dimension_descriptor* %119, i32 0, i32 2
  %123 = getelementptr %dimension_descriptor, %dimension_descriptor* %119, i32 0, i32 3
  store i32 1, i32* %120, align 4
  store i32 1, i32* %121, align 4
  store i32 4, i32* %122, align 4
  %124 = load i32, i32* %122, align 4
  %125 = load i32, i32* %121, align 4
  %126 = sub i32 %124, %125
  %127 = add i32 %126, 1
  store i32 %127, i32* %123, align 4
  %128 = alloca i32, align 4
  store i32 12, i32* %128, align 4
  %129 = getelementptr %array.3, %array.3* %f, i32 0, i32 0
  %130 = load i32, i32* %128, align 4
  %131 = alloca i32, i32 %130, align 4
  store i32* %131, i32** %129, align 8
  %g = alloca %array.4, align 8
  %132 = getelementptr %array.4, %array.4* %g, i32 0, i32 1
  store i32 0, i32* %132, align 4
  %133 = getelementptr %array.4, %array.4* %g, i32 0, i32 2
  %134 = alloca i32, align 4
  store i32 2, i32* %134, align 4
  %135 = load i32, i32* %134, align 4
  %136 = alloca %dimension_descriptor, i32 %135, align 8
  store %dimension_descriptor* %136, %dimension_descriptor** %133, align 8
  %137 = load %dimension_descriptor*, %dimension_descriptor** %133, align 8
  %138 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %137, i32 0
  %139 = getelementptr %dimension_descriptor, %dimension_descriptor* %138, i32 0, i32 0
  %140 = getelementptr %dimension_descriptor, %dimension_descriptor* %138, i32 0, i32 1
  %141 = getelementptr %dimension_descriptor, %dimension_descriptor* %138, i32 0, i32 2
  %142 = getelementptr %dimension_descriptor, %dimension_descriptor* %138, i32 0, i32 3
  store i32 1, i32* %139, align 4
  store i32 1, i32* %140, align 4
  store i32 5, i32* %141, align 4
  %143 = load i32, i32* %141, align 4
  %144 = load i32, i32* %140, align 4
  %145 = sub i32 %143, %144
  %146 = add i32 %145, 1
  store i32 %146, i32* %142, align 4
  %147 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %137, i32 1
  %148 = getelementptr %dimension_descriptor, %dimension_descriptor* %147, i32 0, i32 0
  %149 = getelementptr %dimension_descriptor, %dimension_descriptor* %147, i32 0, i32 1
  %150 = getelementptr %dimension_descriptor, %dimension_descriptor* %147, i32 0, i32 2
  %151 = getelementptr %dimension_descriptor, %dimension_descriptor* %147, i32 0, i32 3
  store i32 1, i32* %148, align 4
  store i32 1, i32* %149, align 4
  store i32 2, i32* %150, align 4
  %152 = load i32, i32* %150, align 4
  %153 = load i32, i32* %149, align 4
  %154 = sub i32 %152, %153
  %155 = add i32 %154, 1
  store i32 %155, i32* %151, align 4
  %156 = alloca i32, align 4
  store i32 10, i32* %156, align 4
  %157 = getelementptr %array.4, %array.4* %g, i32 0, i32 0
  %158 = load i32, i32* %156, align 4
  %159 = alloca i1, i32 %158, align 1
  store i1* %159, i1** %157, align 8
  %h = alloca %array.5, align 8
  %160 = getelementptr %array.5, %array.5* %h, i32 0, i32 1
  store i32 0, i32* %160, align 4
  %161 = getelementptr %array.5, %array.5* %h, i32 0, i32 2
  %162 = alloca i32, align 4
  store i32 3, i32* %162, align 4
  %163 = load i32, i32* %162, align 4
  %164 = alloca %dimension_descriptor, i32 %163, align 8
  store %dimension_descriptor* %164, %dimension_descriptor** %161, align 8
  %165 = load %dimension_descriptor*, %dimension_descriptor** %161, align 8
  %166 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %165, i32 0
  %167 = getelementptr %dimension_descriptor, %dimension_descriptor* %166, i32 0, i32 0
  %168 = getelementptr %dimension_descriptor, %dimension_descriptor* %166, i32 0, i32 1
  %169 = getelementptr %dimension_descriptor, %dimension_descriptor* %166, i32 0, i32 2
  %170 = getelementptr %dimension_descriptor, %dimension_descriptor* %166, i32 0, i32 3
  store i32 1, i32* %167, align 4
  store i32 1, i32* %168, align 4
  store i32 2, i32* %169, align 4
  %171 = load i32, i32* %169, align 4
  %172 = load i32, i32* %168, align 4
  %173 = sub i32 %171, %172
  %174 = add i32 %173, 1
  store i32 %174, i32* %170, align 4
  %175 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %165, i32 1
  %176 = getelementptr %dimension_descriptor, %dimension_descriptor* %175, i32 0, i32 0
  %177 = getelementptr %dimension_descriptor, %dimension_descriptor* %175, i32 0, i32 1
  %178 = getelementptr %dimension_descriptor, %dimension_descriptor* %175, i32 0, i32 2
  %179 = getelementptr %dimension_descriptor, %dimension_descriptor* %175, i32 0, i32 3
  store i32 1, i32* %176, align 4
  store i32 1, i32* %177, align 4
  store i32 3, i32* %178, align 4
  %180 = load i32, i32* %178, align 4
  %181 = load i32, i32* %177, align 4
  %182 = sub i32 %180, %181
  %183 = add i32 %182, 1
  store i32 %183, i32* %179, align 4
  %184 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %165, i32 2
  %185 = getelementptr %dimension_descriptor, %dimension_descriptor* %184, i32 0, i32 0
  %186 = getelementptr %dimension_descriptor, %dimension_descriptor* %184, i32 0, i32 1
  %187 = getelementptr %dimension_descriptor, %dimension_descriptor* %184, i32 0, i32 2
  %188 = getelementptr %dimension_descriptor, %dimension_descriptor* %184, i32 0, i32 3
  store i32 1, i32* %185, align 4
  store i32 1, i32* %186, align 4
  store i32 4, i32* %187, align 4
  %189 = load i32, i32* %187, align 4
  %190 = load i32, i32* %186, align 4
  %191 = sub i32 %189, %190
  %192 = add i32 %191, 1
  store i32 %192, i32* %188, align 4
  %193 = alloca i32, align 4
  store i32 24, i32* %193, align 4
  %194 = getelementptr %array.5, %array.5* %h, i32 0, i32 0
  %195 = load i32, i32* %193, align 4
  %196 = alloca float, i32 %195, align 4
  store float* %196, float** %194, align 8
  %i = alloca %array.6, align 8
  %197 = getelementptr %array.6, %array.6* %i, i32 0, i32 1
  store i32 0, i32* %197, align 4
  %198 = getelementptr %array.6, %array.6* %i, i32 0, i32 2
  %199 = alloca i32, align 4
  store i32 3, i32* %199, align 4
  %200 = load i32, i32* %199, align 4
  %201 = alloca %dimension_descriptor, i32 %200, align 8
  store %dimension_descriptor* %201, %dimension_descriptor** %198, align 8
  %202 = load %dimension_descriptor*, %dimension_descriptor** %198, align 8
  %203 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %202, i32 0
  %204 = getelementptr %dimension_descriptor, %dimension_descriptor* %203, i32 0, i32 0
  %205 = getelementptr %dimension_descriptor, %dimension_descriptor* %203, i32 0, i32 1
  %206 = getelementptr %dimension_descriptor, %dimension_descriptor* %203, i32 0, i32 2
  %207 = getelementptr %dimension_descriptor, %dimension_descriptor* %203, i32 0, i32 3
  store i32 1, i32* %204, align 4
  store i32 1, i32* %205, align 4
  store i32 3, i32* %206, align 4
  %208 = load i32, i32* %206, align 4
  %209 = load i32, i32* %205, align 4
  %210 = sub i32 %208, %209
  %211 = add i32 %210, 1
  store i32 %211, i32* %207, align 4
  %212 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %202, i32 1
  %213 = getelementptr %dimension_descriptor, %dimension_descriptor* %212, i32 0, i32 0
  %214 = getelementptr %dimension_descriptor, %dimension_descriptor* %212, i32 0, i32 1
  %215 = getelementptr %dimension_descriptor, %dimension_descriptor* %212, i32 0, i32 2
  %216 = getelementptr %dimension_descriptor, %dimension_descriptor* %212, i32 0, i32 3
  store i32 1, i32* %213, align 4
  store i32 1, i32* %214, align 4
  store i32 4, i32* %215, align 4
  %217 = load i32, i32* %215, align 4
  %218 = load i32, i32* %214, align 4
  %219 = sub i32 %217, %218
  %220 = add i32 %219, 1
  store i32 %220, i32* %216, align 4
  %221 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %202, i32 2
  %222 = getelementptr %dimension_descriptor, %dimension_descriptor* %221, i32 0, i32 0
  %223 = getelementptr %dimension_descriptor, %dimension_descriptor* %221, i32 0, i32 1
  %224 = getelementptr %dimension_descriptor, %dimension_descriptor* %221, i32 0, i32 2
  %225 = getelementptr %dimension_descriptor, %dimension_descriptor* %221, i32 0, i32 3
  store i32 1, i32* %222, align 4
  store i32 1, i32* %223, align 4
  store i32 3, i32* %224, align 4
  %226 = load i32, i32* %224, align 4
  %227 = load i32, i32* %223, align 4
  %228 = sub i32 %226, %227
  %229 = add i32 %228, 1
  store i32 %229, i32* %225, align 4
  %230 = alloca i32, align 4
  store i32 36, i32* %230, align 4
  %231 = getelementptr %array.6, %array.6* %i, i32 0, i32 0
  %232 = load i32, i32* %230, align 4
  %233 = alloca i32, i32 %232, align 4
  store i32* %233, i32** %231, align 8
  %j = alloca %array.7, align 8
  %234 = getelementptr %array.7, %array.7* %j, i32 0, i32 1
  store i32 0, i32* %234, align 4
  %235 = getelementptr %array.7, %array.7* %j, i32 0, i32 2
  %236 = alloca i32, align 4
  store i32 3, i32* %236, align 4
  %237 = load i32, i32* %236, align 4
  %238 = alloca %dimension_descriptor, i32 %237, align 8
  store %dimension_descriptor* %238, %dimension_descriptor** %235, align 8
  %239 = load %dimension_descriptor*, %dimension_descriptor** %235, align 8
  %240 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %239, i32 0
  %241 = getelementptr %dimension_descriptor, %dimension_descriptor* %240, i32 0, i32 0
  %242 = getelementptr %dimension_descriptor, %dimension_descriptor* %240, i32 0, i32 1
  %243 = getelementptr %dimension_descriptor, %dimension_descriptor* %240, i32 0, i32 2
  %244 = getelementptr %dimension_descriptor, %dimension_descriptor* %240, i32 0, i32 3
  store i32 1, i32* %241, align 4
  store i32 1, i32* %242, align 4
  store i32 5, i32* %243, align 4
  %245 = load i32, i32* %243, align 4
  %246 = load i32, i32* %242, align 4
  %247 = sub i32 %245, %246
  %248 = add i32 %247, 1
  store i32 %248, i32* %244, align 4
  %249 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %239, i32 1
  %250 = getelementptr %dimension_descriptor, %dimension_descriptor* %249, i32 0, i32 0
  %251 = getelementptr %dimension_descriptor, %dimension_descriptor* %249, i32 0, i32 1
  %252 = getelementptr %dimension_descriptor, %dimension_descriptor* %249, i32 0, i32 2
  %253 = getelementptr %dimension_descriptor, %dimension_descriptor* %249, i32 0, i32 3
  store i32 1, i32* %250, align 4
  store i32 1, i32* %251, align 4
  store i32 2, i32* %252, align 4
  %254 = load i32, i32* %252, align 4
  %255 = load i32, i32* %251, align 4
  %256 = sub i32 %254, %255
  %257 = add i32 %256, 1
  store i32 %257, i32* %253, align 4
  %258 = getelementptr inbounds %dimension_descriptor, %dimension_descriptor* %239, i32 2
  %259 = getelementptr %dimension_descriptor, %dimension_descriptor* %258, i32 0, i32 0
  %260 = getelementptr %dimension_descriptor, %dimension_descriptor* %258, i32 0, i32 1
  %261 = getelementptr %dimension_descriptor, %dimension_descriptor* %258, i32 0, i32 2
  %262 = getelementptr %dimension_descriptor, %dimension_descriptor* %258, i32 0, i32 3
  store i32 1, i32* %259, align 4
  store i32 1, i32* %260, align 4
  store i32 2, i32* %261, align 4
  %263 = load i32, i32* %261, align 4
  %264 = load i32, i32* %260, align 4
  %265 = sub i32 %263, %264
  %266 = add i32 %265, 1
  store i32 %266, i32* %262, align 4
  %267 = alloca i32, align 4
  store i32 20, i32* %267, align 4
  %268 = getelementptr %array.7, %array.7* %j, i32 0, i32 0
  %269 = load i32, i32* %267, align 4
  %270 = alloca i1, i32 %269, align 1
  store i1* %270, i1** %268, align 8
  ret i32 0
}
